scripts/release.sh

```diff
@@ -43,31 +41,58 @@ fi
 develop_branch="develop"
 main_branch="master"
 
-# Stash any local changes to avoid errors during checkout
-git status
-prompt_user "Would you like to stash any local changes"
-if [[ $? == 1 ]]; then
-    # Stash local changes to prevent issues with checkout
-    git stash
-    echo "use \"git stash pop\" when this script is complete to restore your changes"
+# Checkout develop - so we can update versions.
+# We need to checkout the branch before we start modifying files.
+current_branch=$(git branch --show-current)
+if [ $current_branch == $develop_branch ]; then
+    echo
+    echo "$develop_branch is already checked out."
 else
-    # Dump local changes to prevent issues with checkout
-    git reset --hard
+    echo
+    echo "Checkout of $develop_branch is required for some steps"
+    prompt_user "Would you like to automatically checkout $develop_branch"
+    if [[ $? == 1 ]]; then
+
+        # Stash any local changes to avoid errors during checkout
+        changes=$(git diff-files)
+        if [ ! -z "$changes" ]; then   
+            echo "There are unstaged changes in these files"
+            echo $changes
+            prompt_user "Would you like to stash these local changes before checkout of $develop_branch"
```

Comment:
<img width="720" alt="Screen Shot 2022-08-02 at 2 12 46 PM" src="https://user-images.githubusercontent.com/1767063/182444911-cbc2f10d-f356-4b0d-9ed1-ad33f7793147.png">

After saying no to this prompt, this is what I saw. Don't know if it's expected.

Reply:
This is expected.  The prompt is for stashing changes.  After stashing changes there is a checkout.  This is the checkout output.  I can maybe add some message in-between to give the user a sense that the "no" option did as expected.

Reply:
More accurately, a pull, and then a checkout of develop.